packages/auth-construct/src/construct.ts

```diff
@@ -900,132 +902,209 @@ export class AmplifyAuth
+          policy.requireUppercase && requirements.push('REQUIRES_UPPERCASE');
+          policy.requireSymbols && requirements.push('REQUIRES_SYMBOLS');
+          return JSON.stringify(requirements);
+        }
+        return undefined;
+      },
+    });
 
-      if (requirements.length > 0) {
-        output.passwordPolicyRequirements = JSON.stringify(requirements);
-      }
-    }
+    // extract the MFA configuration setting from the UserPool resource
+    output.mfaConfiguration = Lazy.string({
+      produce: () => {
+        return cfnUserPool.mfaConfiguration ?? 'OFF';
+      },
+    });
+    // extract the MFA types from the UserPool resource
+    output.mfaTypes = Lazy.string({
+      produce: () => {
+        const mfaTypes: string[] = [];
+        (cfnUserPool.enabledMfas ?? []).forEach((type) => {
+          if (type === 'SMS_MFA') {
+            mfaTypes.push('SMS');
+          }
+          if (type === 'SOFTWARE_TOKEN_MFA') {
+            mfaTypes.push('TOTP');
+          }
+        });
+        return JSON.stringify(mfaTypes);
+      },
+    });
 
-    if (this.computedUserPoolProps.mfa) {
-      output.mfaConfiguration = this.computedUserPoolProps.mfa;
+    // extract social providers from UserPool resource
+    output.socialProviders = Lazy.string({
+      produce: () => {
+        const outputProviders = [];
+        const userPoolProviders = this.resources.userPool.identityProviders;
+        if (userPoolProviders) {
+          for (const provider of userPoolProviders) {
+            const providerType =
+              provider.node['_children']['Resource']['providerType'];
 
-      const mfaTypes = [];
-      if (this.computedUserPoolProps.mfaSecondFactor?.otp) {
-        mfaTypes.push('TOTP');
-      }
-      if (this.computedUserPoolProps.mfaSecondFactor?.sms) {
-        mfaTypes.push('SMS');
-      }
-      if (mfaTypes.length > 0) {
-        output.mfaTypes = JSON.stringify(mfaTypes);
-      }
-    }
+            if (providerType === 'Google') {
+              outputProviders.push('GOOGLE');
+            }
+            if (providerType === 'Facebook') {
+              outputProviders.push('FACEBOOK');
+            }
+            if (providerType === 'SignInWithApple') {
+              outputProviders.push('SIGN_IN_WITH_APPLE');
+            }
+            if (providerType === 'LoginWithAmazon') {
+              outputProviders.push('LOGIN_WITH_AMAZON');
+            }
+            if (providerType === 'OIDC') {
+              outputProviders.push(
+                provider.node['_children']['Resource']['providerName']
+              );
+            }
+            if (providerType === 'SAML') {
+              outputProviders.push(
+                provider.node['_children']['Resource']['providerName']
+              );
+            }
+          }
+          if (outputProviders.length > 0) {
+            return JSON.stringify(outputProviders);
+          }
+        }
+        return undefined;
```

Comment:
Pls move the `return undefined;` to top and change it to `return`

Reply:
updated